#!/usr/bin/env python
PACKAGE = "dvs_ros_driver"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("streaming_rate", int_t, 0, "integration time", 30, 1, 500)

gen.add("cas", int_t, 0, "cas", 1992, 1992, 1992)
gen.add("injGnd", int_t, 0, "injGnd", 1108364, 1108364, 1108364)
gen.add("reqPd", int_t, 0, "reqPd", 16777215, 16777215, 16777215)
gen.add("puX", int_t, 0, "puX", 8159221, 8159221, 8159221)
gen.add("diffOff", int_t, 0, "diffOff", 132, 20, 500)
gen.add("req", int_t, 0, "req", 309590, 309590, 309590)
gen.add("refr", int_t, 0, "refr", 969, 969, 969)
gen.add("puY", int_t, 0, "puY", 16777215, 16777215, 2000000)
gen.add("diffOn", int_t, 0, "diffOn", 209996, 120000, 700000)
gen.add("diff", int_t, 0, "diff", 13125, 13125, 13125)
gen.add("foll", int_t, 0, "foll", 271, 271, 271)
gen.add("pr", int_t, 0, "pr", 217, 217, 217)

exit(gen.generate(PACKAGE, "dvs_ros_driver", "DVS_ROS_Driver"))
